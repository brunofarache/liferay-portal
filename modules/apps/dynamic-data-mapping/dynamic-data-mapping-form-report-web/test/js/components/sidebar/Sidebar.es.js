/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import Sidebar from '../../../../src/main/resources/META-INF/resources/js/components/sidebar/Sidebar.es';
import SidebarContext from '../../../../src/main/resources/META-INF/resources/js/components/sidebar/SidebarContext.es';

const field = {
    icon: "text",
	label: "Text",
	name: "Text",
	options: {Option: "Option"},
	title: "Text",
	type: "text",
}

const sidebarState = {
	field,
	isOpen: true,
	totalEntries: 6
}

const context = {
	...sidebarState,
	formReportRecordsFieldValuesURL: '/',
	portletNamespace: '_com_liferay_dynamic_data_mapping_form_report_web_portlet_DDMFormReportPortlet_',
	toggleSidebar: () => !sidebarState.isOpen
}


describe('SidebarContext', async () => {
	afterEach(cleanup);

	it('renders', () => {
		fetch.mockResponseOnce(JSON.stringify(LIST));

		// const {asFragment} = render(
		// 	<SidebarContext value={context}>
		// 		<Sidebar />
		// 	</SidebarContext>
		// );

		// expect(asFragment()).toMatchSnapshot();
	});
});

const LIST = ['name1', 'name2', 'name3', 'name4', 'name5', 'name6'];
