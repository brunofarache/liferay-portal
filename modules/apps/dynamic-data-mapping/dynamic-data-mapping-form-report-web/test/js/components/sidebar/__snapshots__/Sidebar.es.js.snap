// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders loading animation while requisition is loading 1`] = `
<DocumentFragment>
  <div
    class="sidebar-container "
  >
    <div
      class="sidebar sidebar-light"
    >
      <nav
        class="component-tbar tbar"
      >
        <div
          class="container-fluid"
        >
          <ul
            class="tbar-nav"
          >
            <li
              class="tbar-item"
            >
              <div
                class="icon"
              >
                <svg
                  class="lexicon-icon lexicon-icon-text"
                  role="presentation"
                >
                  <use
                    xlink:href="#text"
                  />
                </svg>
              </div>
            </li>
            <li
              class="tbar-item tbar-item-expand"
            >
              <div
                class="tbar-section"
              >
                <div
                  class="field-info"
                >
                  <p
                    class="title"
                  >
                    fieldName
                  </p>
                  <p
                    class="description"
                  >
                    5 entries
                  </p>
                </div>
              </div>
            </li>
            <li
              class="tbar-item"
            >
              <button
                class="btn btn-unstyled"
                type="button"
              >
                <svg
                  class="lexicon-icon lexicon-icon-times-small close-button"
                  role="presentation"
                >
                  <use
                    xlink:href="#times-small"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </div>
      </nav>
      <div
        class="sidebar-body"
      >
        <div
          class="align-items-center d-flex loading-wrapper"
        >
          <span
            aria-hidden="true"
            class="loading-animation"
          />
        </div>
        <ul
          class="entries-list"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
